acceptance/tests/facts/networking_facts.rb

```diff
@@ -0,0 +1,80 @@
+test_name 'FACT-1361 - C59029 networking facts should be fully populated'
+
+#
+# This test is intended to ensure that networking facts resolves
+# as expected in AIO across supported platforms.
+#
+
+skip_test "Networking fact test is confined to AIO" if @options[:type] != 'aio'
+
+@ip_regex = /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/
+
+expected_networking = {
+    "networking.dhcp"     => @ip_regex,
+    "networking.ip"       => @ip_regex,
+    "networking.ip6"      => /[a-f0-9]+:+/,
+    "networking.mac"      => /[a-f0-9]{2}:/,
+    "networking.mtu"      => /\d+/,
+    "networking.netmask"  => /\d+\.\d+\.\d+\.\d+/,
+    "networking.netmask6" => /[a-f0-9]+:/,
```

Comment:
The hex characters do come in groups for four, and there are 8 groups, but there is a short hand that allows skipping one run of contiguous 0s.  An address may look like 2001:0db8::1234:5678, to mean 2001:0db8:0000:0000:0000:0000:1234:5678 (zeros between the two contiguous colons).  This makes a recognizing regexp very nasty (see http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses)